pkg/function/create_volume_from_snapshot.go

```diff
@@ -125,6 +128,22 @@ func createVolumeFromSnapshot(ctx context.Context, cli kubernetes.Interface, nam
 	return providerList, nil
 }
 
+func addPVProvisionedByAnnotation(annotations map[string]string, provider blockstorage.Provider) map[string]string {
+	if annotations == nil {
```

Comment:
Would we rather defer this check until after storageType check so that we don't change a nil annotations into an empty map for other providers?

Just asking, though it might not matter in this code.

Reply:
yeah i also think if we are here it would mean that we are going to add those annotations, that is the reason I didn't have that check before. can add it, if we think thats better.